File: ASSAULT_COURSE/ASSAULT_COURSE/src/com/codemanship/refactoring/assaultcourse/Customer.java
package com.codemanship.refactoring.assaultcourse;

import java.util.ArrayList;
import java.util.List;

public class Customer {
	
	private List<Video> rentals = new ArrayList<Video>();
	private Address address;
	private final String dateOfBirth;
	private final String name;


	public Customer(String name, Address address, String dateOfBirth) {
		this.name = name;
		this.address = address;
		this.dateOfBirth = dateOfBirth;
	}

	public String getDateOfBirth() {
		return dateOfBirth;
	}
	
	public List<Video> getRentedVideos() {
		return rentals;
	}

	public void addRental(Video video) {
		rentals.add(video);
	}

	public String getSummary() {
		return name + ", " + address.getHouse() + " " + address.getStreet() + ", " 
				+ address.getCity() + ", " + address.getPostcode();
	}

}

File: ruby/assault_course/customer.rb
class Customer
  attr_reader :date_of_birth
  attr_reader :rentals

  def initialize(name, address, date_of_birth)
    @name = name
    @address = address
    @date_of_birth = date_of_birth
    @rentals = []
  end

  def add_rental(movie)
    @rentals << movie
  end

  def summary
    "#{@name}, #{@address.house} #{@address.street}, #{@address.city}, #{@address.zipcode}"
  end

end
